apps/lib/chain_pusher/solana.go

```diff
@@ -0,0 +1,411 @@
+			Hi: quantizedPriceBigInt.Rsh(quantizedPriceBigInt, 64).Uint64(),
+		}
+
+		publisherMerkleRootBytes, err := hexStringToByteArray(priceUpdate.StorkSignedPrice.PublisherMerkleRoot)
+		if err != nil {
+			return fmt.Errorf("failed to convert PublisherMerkleRoot: %w", err)
+		}
+		var publisherMerkleRoot [32]uint8
+		copy(publisherMerkleRoot[:], publisherMerkleRootBytes)
+
+		valueComputeAlgHashBytes, err := hexStringToByteArray(priceUpdate.StorkSignedPrice.StorkCalculationAlg.Checksum)
+		if err != nil {
+			return fmt.Errorf("failed to convert ValueComputeAlgHash: %w", err)
+		}
+
+		var valueComputeAlgHash [32]uint8
+		copy(valueComputeAlgHash[:], valueComputeAlgHashBytes)
+
+		rBytes, err := hexStringToByteArray(priceUpdate.StorkSignedPrice.TimestampedSignature.Signature.R)
+		if err != nil {
+			return fmt.Errorf("failed to convert R: %w", err)
+		}
+		var r [32]uint8
+		copy(r[:], rBytes)
+
+		sBytes, err := hexStringToByteArray(priceUpdate.StorkSignedPrice.TimestampedSignature.Signature.S)
+		if err != nil {
+			return fmt.Errorf("failed to convert S: %w", err)
+		}
+		var s [32]uint8
+		copy(s[:], sBytes)
+
+		vBytes, err := hexStringToByteArray(priceUpdate.StorkSignedPrice.TimestampedSignature.Signature.V)
+		if err != nil {
+			return fmt.Errorf("failed to convert V: %w", err)
+		}
+		v := uint8(vBytes[0])
+
+		updateData := contract.TemporalNumericValueEvmInput{
+			TemporalNumericValue: contract.TemporalNumericValue{
+				TimestampNs:    uint64(priceUpdate.StorkSignedPrice.TimestampedSignature.Timestamp),
+				QuantizedValue: quantizedPrice,
+			},
+			Id:                  assetId,
+			PublisherMerkleRoot: publisherMerkleRoot,
+			ValueComputeAlgHash: valueComputeAlgHash,
+			R:                   r,
+			S:                   s,
+			V:                   v,
+			TreasuryId:          treasuryId,
+		}
+
+		instruction, err := contract.NewUpdateTemporalNumericValueEvmInstruction(
+			updateData,
+			configAccount,
+			treasuryAccount,
+			feedAccount,
+			sci.payer.PublicKey(),
+			solana.SystemProgramID,
+		).ValidateAndBuild()
+		if err != nil {
+			return fmt.Errorf("failed to build instruction: %w", err)
+		}
+		instructions = append(instructions, instruction)
+
+	}
+
+	recentBlockHash, err := sci.client.GetRecentBlockhash(context.Background(), rpc.CommitmentFinalized)
+	if err != nil {
+		return fmt.Errorf("failed to get recent blockhash: %w", err)
+	}
+
+	tx, err := solana.NewTransaction(
+		instructions,
+		recentBlockHash.Value.Blockhash,
+		solana.TransactionPayer(sci.payer.PublicKey()),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create transaction: %w", err)
+	}
+
+	_, err = tx.Sign(
+		func(key solana.PublicKey) *solana.PrivateKey {
+			if key == sci.payer.PublicKey() {
+				return &sci.payer
+			}
+			return nil
+		})
+	if err != nil {
+		return fmt.Errorf("failed to sign transaction: %w", err)
+	}
+
+	sig, err := confirm.SendAndConfirmTransaction(context.Background(), sci.client, sci.wsClient, tx)
```

Comment:
Is it slow to confirm the transaction? Maybe only confirm if given a certain configuration to go faster?

Reply:
Yeah confirming is pretty slow, however it doesn't slow down the sending of the transaction. This isn't actually blocking other batches from going out from inside batchpushtocontract. I do see that batchpushtocontract blocks the select statement in Run() tho. I think I'd prefer spinning up a separate go routine for the batch push to contract call in the select statement in run. That wait we can confirm by default without actually slowing anything down. How do you feel about that?